ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/stddev/StandardDeviationSampUdaf.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.stddev;
+
+import io.confluent.ksql.function.udaf.TableUdaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import io.confluent.ksql.util.KsqlConstants;
+import java.util.function.BiFunction;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+@UdafDescription(name = "STD_DEV_SAMP",
+    description = "Returns the sample standard deviation of the column. "
+        +
+        "Applicable only to numeric types.",
+    author = KsqlConstants.CONFLUENT_AUTHOR
+)
+public final class StandardDeviationSampUdaf {
+
+  private static final String COUNT = "COUNT";
+  private static final String SUM = "SUM";
+  private static final String M2 = "M2";
+  private static final Schema STRUCT_LONG = SchemaBuilder.struct().optional()
+      .field(SUM, Schema.OPTIONAL_INT64_SCHEMA)
+      .field(COUNT, Schema.OPTIONAL_INT32_SCHEMA)
+      .field(M2, Schema.OPTIONAL_FLOAT64_SCHEMA)
+      .build();
+
+  private static final Schema STRUCT_INT = SchemaBuilder.struct().optional()
+      .field(SUM, Schema.OPTIONAL_INT32_SCHEMA)
+      .field(COUNT, Schema.OPTIONAL_INT32_SCHEMA)
+      .field(M2, Schema.OPTIONAL_FLOAT64_SCHEMA)
+      .build();
+
+  private static final Schema STRUCT_DOUBLE = SchemaBuilder.struct().optional()
+      .field(SUM, Schema.OPTIONAL_FLOAT64_SCHEMA)
+      .field(COUNT, Schema.OPTIONAL_INT32_SCHEMA)
+      .field(M2, Schema.OPTIONAL_FLOAT64_SCHEMA)
+      .build();
+
+  private StandardDeviationSampUdaf() {
+  }
+
+  @UdafFactory(description = "Compute sample standard deviation of column with type Long.",
+      aggregateSchema = "STRUCT<SUM bigint, COUNT integer, M2 double>")
+  public static TableUdaf<Long, Struct, Double> stdDevLong() {
+    return getStdDevImplementation(
+        0L,
+        STRUCT_LONG,
+        (agg, newValue) -> newValue + agg.getInt64(SUM),
+        (agg, newValue) ->
+            Double.valueOf(newValue * (agg.getInt32(COUNT) + 1) - (agg.getInt64(SUM) + newValue)),
+        (agg, count) -> agg.getFloat64(M2) / (count - 1),
```

Comment:
This operation is the same on all 3 type s(long, int, double). Can ` agg.getFloat64(M2) / (count - 1)` be called directly from the `map` function instead of calling the BiFunction?

Reply:
It definitely can